### PR TITLE
Sync from moveit1

### PR DESCRIPTION
**Pull request by tylerjw**
Merging this should fix the history and doing a log against main should work agan.

### CODE REVIEW

File: .github/workflows/cross_platform_ci.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
+name: MoveIt core cross-platform RoboStack build
```
Comment (tylerjw) [inline .github/workflows/cross_platform_ci.yml:1]: I'm not sure we want to sync this, @henningkayser what do I do if I want to drop this commit?  Should I do an interactive rebase and delete the commit this came from?

File: moveit_core/kinematics_base/CMakeLists.txt [hunk 2]
```diff
@@ -6,6 +6,10 @@ include(GenerateExportHeader)
 generate_export_header(${MOVEIT_LIB_NAME})
 target_include_directories(${MOVEIT_LIB_NAME} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
 
+include(GenerateExportHeader)
```
Comment (henningkayser) [inline moveit_core/kinematics_base/CMakeLists.txt:9]: This is duplicate with the lines above

File: moveit_core/package.xml [hunk 3]
```diff
@@ -30,6 +30,8 @@
   <depend>boost</depend>
   <depend>eigen</depend>
   <depend>bullet</depend>
+  <depend condition="$ROS_DISTRO == noetic">python3</depend>
```
Comment (henningkayser) [inline moveit_core/package.xml:33]: Remove the noetic condition and line below, just depend on `python3` directly

File: moveit_core/python/tools/CMakeLists.txt [hunk 4]
```diff
@@ -1,10 +1,17 @@
 set(MOVEIT_LIB_NAME moveit_python_tools)
 
+find_package(Python COMPONENTS Interpreter Development QUIET)
+
 add_library(${MOVEIT_LIB_NAME}
   src/pybind_rosmsg_typecasters.cpp
 )
+<<<<<<< HEAD
```
Comment (henningkayser) [inline moveit_core/python/tools/CMakeLists.txt:8]: Unresolved conflict

File: moveit_core/collision_distance_field/src/collision_env_distance_field.cpp [hunk 5]
```diff
@@ -50,11 +50,13 @@
 
 namespace collision_detection
 {
+namespace
+{
 const rclcpp::Logger CollisionEnvDistanceField::LOGGER =
```
Comment (henningkayser) [inline moveit_core/collision_distance_field/src/collision_env_distance_field.cpp:55]: this LOGGER must be outside the anonymous namespace

**Comment by tylerjw** [comment]
Closing as this was fixed separately.

**Pull request closed without merging**